### PR TITLE
feat!: support for building argus monorepo all in the same workflow

### PR DESCRIPTION
**Pull request by hspitzley-czi**
Example of this being used: https://github.com/chanzuckerberg/bifrost/pull/18

### CODE REVIEW

File: .github/actions/argus-builder/build-prep/action.yml [hunk 1]
```diff
@@ -29,37 +29,19 @@ outputs:
   should_build:
     description: Whether the job should run
     value: ${{ steps.final_check.outputs.should_build }}
+  base:
+    description: The base branch or commit to compare against
+    value: ${{ steps.force_push.outputs.base }}
```
Comment (jakeyheath) [inline .github/actions/argus-builder/build-prep/action.yml:34]: maybe `base_ref` might be a more descriptive name. `base` is an overloaded term we use a lot

File: .github/actions/check-branch-match/action.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  using: composite
+  steps:
+    - name: Check for matching branch
+      id: branch_filter
+      uses: actions/github-script@v7
+      with:
+        script: |
+          function wildcardMatch(text, pattern) {
+            const regexPattern =
+            new RegExp('^' + pattern.replace(/\?/g, '.').replace(/\*/g, '.*') + '$');
```
Comment (jakeyheath) [inline .github/actions/check-branch-match/action.yaml:33]: is this to simulate glob patterns to match the dorny action?

Comment (jakeyheath) [reply .github/actions/check-branch-match/action.yaml:33]: mind adding a short comment about why the regex works the way it does?

File: .github/actions/validate-json-schema/action.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+name: argus-docker-build-prep
+description: Prepare for building a Docker Image for Argus
```
Comment (jakeyheath) [inline .github/actions/validate-json-schema/action.yml:2]: update

File: .github/actions/validate-json-schema/action.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  using: composite
+  steps:
+    - uses: actions/setup-node@v4
+      with:
+        node-version: '20'
+    - run: npm install ajv@8
+      shell: bash
+    - uses: actions/github-script@v7
+      with:
+        script: |
```
Comment (jakeyheath) [inline .github/actions/validate-json-schema/action.yml:22]: smart, this will be nice

**Review by jakeyheath — APPROVED** [review]
some small nits

File: .github/actions/check-branch-match/action.yaml (hunk 2 above)
Comment (hspitzley-czi) [reply .github/actions/check-branch-match/action.yaml:33]: i originally added this to match how github path filters work

**Review by jakeyheath — APPROVED** [review]

**Pull request merged**